### PR TITLE
Install react-dom & add babel plugin

### PR DESCRIPTION
**Pull request by tamsingreen**
**Is your Pull Request request related to another [issue](https://github.com/bbc/digital-paper-edit-storybook/issues) in this repository ?**      
https://github.com/bbc/digital-paper-edit-stt-proxy/issues/40

**Describe what the PR does**
Add missing react-dom dependency.
Adds all react-related dependencies to peerDependencies.
When `npm run link:setup` is run, the react-related dependencies are removed, so there's no clash of React versions when running the firebase app.
Copies package.json to the dist directory when linked or published and removes react-related dependencies.

**State whether the PR is ready for review or whether it needs extra work**    
RFR

**Steps to test**
Ensure storybook and firebase projects are both using Node 12.18.4 (`nvm use 12.18.4`).

1. In the storybook repo:
```
rm -rf node_modules && npm i && npm start
```
2. Check the storybook app works as expected. Then Ctrl-C out and run:
```
npm run link:setup
```
Check that ./dist exists, has .js and .js.map files and a package.json without react dependencies (react, react-dom, react-router & react-router dom)

3. In another terminal window, still in the project root, run:
```
npm link
```
4. In the firebase repo, start the app:
```
npm run dev:react
```
Check the app works as expected.

5. To test changes to storybook are reflected in firebase, in the storybook repo, make a change or add a console.log in one of the components, then check that shows in the firebase app.

6. When finished, from the storybook repo, run:
```
npm run unlink
```
This will undo the npm link and install the react deps back into the project.

### CODE REVIEW

**Comment by emettely** [comment]
Jogging my memory - having React Dom actually being a part of the dependencies will cause a problem when the components get wrapped and published.

When you ingest the storybook components, essentially there will be 2 versions of React DOM which is going to cause some odd error messages. https://github.com/bbc/digital-paper-edit-storybook/blob/f9d5323cbe113ea0add61f5f58b96e2a900dcec6/docs/guides/error-invariant-multiple-react-router-dom-09-28.md

File: package.json [hunk 1]
```diff
@@ -6,20 +6,21 @@
...
-    "linked": "npm run clean && mkdir -p dist && npm run cp && npm run dev && npm run cp",
+    "linked": "npm run clean && mkdir -p dist && npm run rm-react-deps && npm run cp && npm run dev",
     "buildw": "webpack --config webpack.config.js --watch",
     "build-storybook": "build-storybook -s public",
     "build-storybook-docs": "build-storybook -s public --docs",
-    "clean": "rimraf dist storybook-static test",
+    "clean": "rm -rf dist",
     "publish-ghpages": "mkdir -p build-storybook && npm run build-storybook && gh-pages -d storybook-static",
-    "cp": "cp package.json ./dist/package.json && cp README.md ./dist/README.md ",
+    "cp": "cp package.dist.json ./dist/package.json && cp README.md ./dist/README.md ",
```
Comment (emettely) [inline package.json:18]: Probably worth adding to the README about what the difference is between the two package files.

**Review by emettely — COMMENTED** [review]
Haven't run it yet, so just leaving a comment until I do :)

**Comment by sarahrainbow** [comment]
Getting this error in my firebase terminal at stage 4  
Screenshot 2021-04-13 at 13 53 12

**Comment by sarahrainbow** [comment]
unlinked and tried the steps again and got a different error this time! 
Screenshot 2021-04-13 at 14 17 13

**Comment by allishultes** [comment]
I've also had some trouble running this, and have tried going back and uninstalling the link, and then setting it up again. I have an error with mismatched versions of React:

Screenshot 2021-04-13 at 15 04 01

**Comment by aniablaziak** [comment]
getting the same Error as Alli at the final step

**Comment by aniablaziak** [comment]
> getting the same Error as Alli at the final step

@allishultes there is a mismatch between 'react-dom' versions between storybook and firebase repos `"react-dom": "^16.13.0"` in firebase and `"react-dom": "^16.12.0",` in storybook, once they are unified the Error is gone but still can't see my logs from Storybook components 🤔

**Comment by tamsingreen** [comment]
Thanks everyone - I think it's back to the drawing board with this one.

**Comment by sarahrainbow** [comment]
Just deleted package-lock file and still encountered the same `Invalid Hook call` error as before 😢

**Comment by aniablaziak** [comment]
merging this as discussed to fix the startup errors.

File: README.md [hunk 2]
```diff
@@ -45,6 +45,23 @@ npm start
...
+```
+
+In another terminal window:
+```sh
+npm run link
+```
+
+When finished, run:
+```sh
+npm run unlink
```
Comment (allishultes) [inline README.md:62]: Does it matter which window this is run in? I ran it in the same window as `npm run link` and that works fine, but just curious.

**Review by allishultes — APPROVED** [review]
This works for me Tamsin!! 💯 💯 💯

**Pull request merged**